core/src/main/scala/cats/Align.scala

```diff
@@ -0,0 +1,87 @@
+package cats
+
+import simulacrum.typeclass
+
+import cats.data.Ior
+
+/**
+ * `Align` supports zipping together structures with different shapes,
+ * holding the results from either or both structures in an `Ior`.
+ *
+ * Must obey the laws in cats.laws.AlignLaws
+ */
+@typeclass trait Align[F[_]] {
+
+  def functor: Functor[F]
```

Comment:
Maybe this was already discussed, but is there a particular reason to make this compositional rather than inherited? Is it because `Align` is a leaf typeclass (a la `Traverse`)?

Reply:
That's exactly the reason :)

Reply:
Shouldn't we try to keep the encodings uniform though? After all, `Traverse` has this same problem, but it still uses subtyping.

Reply:
I think we should move away from it. We're already doing this for `FunctorFilter`, `TraverseFilter`, `Representable` and probably others :)

Reply:
https://github.com/typelevel/cats/issues/2400